src/main/java/edu/cornell/kfs/coa/document/web/struts/CuObjectCodeActivationGlobalMaintenanceDocumentAction.java

```diff
@@ -0,0 +1,55 @@
+package edu.cornell.kfs.coa.document.web.struts;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.struts.action.ActionForm;
+import org.apache.struts.action.ActionForward;
+import org.apache.struts.action.ActionMapping;
+import org.kuali.kfs.kns.web.struts.form.KualiMaintenanceForm;
+import org.kuali.kfs.krad.bo.Note;
+import org.kuali.kfs.krad.util.GlobalVariables;
+import org.kuali.kfs.sys.document.FinancialSystemMaintenanceDocument;
+import org.kuali.rice.kim.api.identity.Person;
+
+import edu.cornell.kfs.coa.businessobject.CuObjectCodeActivationGlobal;
+import edu.cornell.kfs.sys.document.web.struts.CuFinancialMaintenanceDocumentAction;
+
+@SuppressWarnings("deprecation")
+public class CuObjectCodeActivationGlobalMaintenanceDocumentAction extends CuFinancialMaintenanceDocumentAction {
+    protected static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(CuObjectCodeActivationGlobalMaintenanceDocumentAction.class);
+    
+    @Override
+    public ActionForward copy(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response) throws Exception {
```

Comment:
Could this customized copying functionality be moved into other places that fit within the regular maintenance document framework (such as the processAfterCopy() method on the Maintainable impl)? If so, then that would simplify the configuration and allow for potentially removing the customized document JSP file.

Reply:
The parent copy action looks to see if the document is a child of maintenance document, if so it automatically errors.  I don't think we can avoid a new action class

Reply:
It looks like the parent copy action can perform the copy if a non-null "document.documentNumber" request parameter is present. If it's not feasible to add such a parameter setup to the in-document copy button/action, then I'm okay with the using a separate action class and JSP for now.

Reply:
The only way I can think of to set the document.documentNumber is by setting it in an Action class's copy method and then call the parent.  To do that we would still need the custom JSP and struts mapping, so I don't think that would buy us improvement over the current code.  

An option would be to override the KFS maintenance document JSP, and then dynamically set document.documentNumber in the form when it is an object code activation global document, and that seems like a bad course as well.

Reply:
Ok, thanks for checking. I'm okay with the current customization for copying these documents.